Code/Tools/FBuild/FBuildCore/ExeDrivers/Compiler/CompilerDriver_ClangTidy.cpp

```diff
@@ -0,0 +1,94 @@
+// CompilerDriver_ClangTidy.cpp
+//------------------------------------------------------------------------------
+
+// Includes
+//------------------------------------------------------------------------------
+#include "CompilerDriver_ClangTidy.h"
+
+// FBuild
+#include "Tools/FBuild/FBuildCore/FBuild.h"
+#include "Tools/FBuild/FBuildCore/Graph/CompilerNode.h"
+#include "Tools/FBuild/FBuildCore/Graph/ObjectNode.h"
+#include "Tools/FBuild/FBuildCore/Helpers/Args.h"
+#include "Tools/FBuild/FBuildCore/WorkerPool/Job.h"
+
+// CONSTRUCTOR
+//------------------------------------------------------------------------------
+CompilerDriver_ClangTidy::CompilerDriver_ClangTidy() = default;
+
+// DESTRUCTOR
+//------------------------------------------------------------------------------
+CompilerDriver_ClangTidy::~CompilerDriver_ClangTidy() = default;
+
+// ProcessArg_PreprocessorOnly
+//------------------------------------------------------------------------------
+/*virtual*/ bool CompilerDriver_ClangTidy::ProcessArg_PreprocessorOnly( const AString & /*token*/,
+                                                                       size_t & /*index*/,
+                                                                       const AString & /*nextToken*/,
+                                                                       Args & /*outFullArgs*/ ) const
+{
+    ASSERT(false); // ClangTidy can't be used as a preprocessor
```

Comment:
Seems odd to implement these functions but assert.

The tests also seem to set preprocessor options, which seems at odds with this.

Reply:
I see later on that we're using a different compile for the preprocessor.

I think we'll need to make sure this works without needing to do that. If that's required for distributed compilation (assuming clang-tidy can't function as a preprocessor?) we'll have to come up with a good way to make it clear that extra manual config is needed if you want distributed clang-tidy.